### PR TITLE
Stats tab: disable completed level count

### PR DESCRIPTION
**Pull request by caleybrock**
User impact: stats tab no longer has completed level count column, only lines of code.
Warning text from Ryan.
screen shot 2018-05-02 at 10 30 16 am

API Impact: Remove the need to count number of levels when fetching students in the pegasus API. This is a potential cause of slow queries and site downtime this week.

### CODE REVIEW

File: pegasus/helpers/section_api_helpers.rb [hunk 1]
```diff
@@ -609,21 +609,10 @@ def students
...
-
-    level_counts_db = Gatekeeper.allows('use_reporting_db_for_progress', default: false) ? Dashboard.db_reporting_reader : Dashboard.db
-
-    level_counts = level_counts_db[:user_levels].
-      group_and_count(:user_id).
-      where(user_id: student_ids).
-      where("best_result >= #{ActivityConstants::MINIMUM_PASS_RESULT}").
-      all
+    # completed_levels_count is no longer needed on the UI,
+    # but adding this feild to not break anything unexpected.
```
Comment (islemaster) [inline pegasus/helpers/section_api_helpers.rb]: `s/feild/field`

File: pegasus/test/test_section_api_helpers.rb [hunk 2]
```diff
@@ -894,7 +894,7 @@ class SectionApiHelperTest < SequelTestCase
                 secret_picture_path: nil,
                 location: "/v2/users/#{FakeDashboard::STUDENT[:id]}",
                 age: nil,
-                completed_levels_count: 1
+                completed_levels_count: 0
```
Comment (islemaster) [inline pegasus/test/test_section_api_helpers.rb]: Potentially worth a comment in the test as well.

File: pegasus/sites.v3/code.org/public/teacher-dashboard/section_stats.haml [hunk 3]
```diff
@@ -5,6 +5,7 @@ content-type: text/ng-template
 ---
 
 #uitest-stats-tab{style: "padding-left:20px", 'ng-init' => '$emit("stats-table-rendered");'}
+  %div.alert.alert-warning The Completed Level count is temporarily disabled and will return in a few days. In the meantime, you can still access level-by-level student progress on the Progress tab.
```
Comment (islemaster) [inline pegasus/sites.v3/code.org/public/teacher-dashboard/section_stats.haml:8]: Think we should link to a blog post or something?

Comment (caleybrock) [reply pegasus/sites.v3/code.org/public/teacher-dashboard/section_stats.haml:8]: @jeremydstone or @ryansloan do we need a blog post or support article for this?

**Comment by ewjordan** [comment]
As far as future refactorings go, if the primary thing we're looking for on each student is a single number per course that just keeps track of how many levels they've completed, I wonder if it makes sense to start storing a per-student map from course -> progress count as a JSON field? It would require a data migration, and would denormalize our data a bit, but it would avoid potentially lengthy aggregation queries which are O(students * progress_per_student) with pretty random access throughout `user_levels` and turn them into O(students) on the `users` table.

File: pegasus/sites.v3/code.org/public/teacher-dashboard/section_stats.haml (hunk 3 above)
Comment (caleybrock) [reply pegasus/sites.v3/code.org/public/teacher-dashboard/section_stats.haml:8]: Just chatted with Ryan, we think this message is sufficient for these teachers.

**Pull request merged**